examples/src/keyring/raw_rsa/public_private_key_separate.py

```diff
@@ -0,0 +1,122 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+One of the benefits of asymmetric encryption
+is that you can encrypt with just the public key.
+This means that you give someone the ability to encrypt
+without giving them the ability to decrypt.
+
+The raw RSA keyring supports encrypt-only operations
```

Comment:
At least in the Java examples, I'm not sure we'd need both this example and the private_key_only examples. The only difference in Java would be passing the optional private key to the encrypt keyring, which doesn't use it anyway.

Reply:
I think it's important to show the example with them separate because that's a common use-case for RSA. I like showing the private-key-only example to demonstrate that you can still encrypt even if you only provide the private key. They're not drastically different, no, but I think that they're *enough* different, and more importantly it's likely that customers will be doing both, that it's worth showing both.